app/uk/gov/hmrc/agentaccesscontrol/connectors/GovernmentGatewayProxyConnector.scala

```diff
@@ -39,15 +40,17 @@ case class AssignedAgent(allocatedAgentCode: AgentCode) {
 
 }
 
-class GovernmentGatewayProxyConnector(baseUrl: URL, httpPost: HttpPost, auditService: AuditService) {
+class GovernmentGatewayProxyConnector(baseUrl: URL, httpPost: HttpPost, auditService: AuditService) extends HttpAPIMonitor {
   val url: URL = new URL(baseUrl, "/government-gateway-proxy/api/admin/GsoAdminGetAssignedAgents")
 
   def getAssignedSaAgents(utr: SaUtr, agentCode: AgentCode)(implicit hc: HeaderCarrier): Future[Seq[AssignedAgent]] = {
-    httpPost.POSTString(url.toString, body(utr), Seq(CONTENT_TYPE -> XML))
+    monitor("ConsumedAPI-GGW-GetAssignedAgents-POST"){
+      httpPost.POSTString(url.toString, body(utr), Seq(CONTENT_TYPE -> XML))
```

Comment:
Is `httpPost` not a `MonitoredWSHttp`? If so, I suggest just adding the URL pattern to its config. If not, I'd wrap only the request to `monitor`, not the request+logging+parsing, so we'd get a more precise result.